apps/balancer-tools/src/app/apr/(utils)/getBALPriceForDateRange.ts

```diff
@@ -1,52 +1,43 @@
 import { networkFor } from "@bleu-fi/utils";
-import { calculateDaysBetween, dateToEpoch } from "@bleu-fi/utils/date";
-import { eq } from "drizzle-orm";
+import { dateToEpoch } from "@bleu-fi/utils/date";
+import { formatNumber } from "@bleu-fi/utils/formatNumber";
+import { eq, sql } from "drizzle-orm";
 
 import { db } from "#/db";
 import { tokenPrices } from "#/db/schema";
 import { withCache } from "#/lib/cache";
 import { DefiLlamaAPI } from "#/lib/defillama";
 
 const BAL_TOKEN_ADDRESS = "0xba100000625a3754423978a60c9317c58a424e3d";
-const BAL_TOKEN_NETWORK = 1;
 
 /**
  * Calculates the average of an array of numbers.
  */
-const calculateAverage = (arr: number[]) =>
-  arr.reduce((sum, val) => sum + val, 0) / arr.length;
+export async function getBALPriceForDateRange(startAt: Date, endAt: Date) {
+  const result = await db.execute(sql`
 
-export const getBALPriceForDateRange = withCache(
-  async function getBALPriceByRoundFn(
-    startAtTimestamp: number,
-    endAtTimestamp: number
-  ) {
-    const numberOfDays = calculateDaysBetween(startAtTimestamp, endAtTimestamp);
-    const pricePromises = Array.from({ length: numberOfDays }, (_) => {
-      return getTokenPriceByDate(
-        endAtTimestamp,
-        BAL_TOKEN_ADDRESS,
-        BAL_TOKEN_NETWORK
-      );
-    });
-    try {
-      const prices = await Promise.all(pricePromises);
-      return calculateAverage(prices);
-    } catch (error) {
-      // TODO: BAL-782 - Add sentry here
-      // eslint-disable-next-line no-console
-      console.error(
-        `Error fetching BAL price between ${startAtTimestamp} and ${endAtTimestamp} - ${error}`
-      );
-      throw error;
-    }
-  }
-);
+      SELECT
+      AVG(price_usd) AS average_price_usd
+    FROM
+      (SELECT
+        price_usd
+       FROM
+        token_prices
+       WHERE
+       token_address = '${sql.raw(BAL_TOKEN_ADDRESS)}'
+       AND timestamp >= '${sql.raw(startAt.toISOString())}'
+       AND timestamp <= '${sql.raw(endAt.toISOString())}'
+      ) AS subquery
+  `);
+  //TODO when there is no price on db
```

Comment:
thinking about using the old version as a fallback and insert the result on the db

Reply:
lgtm